File: leetcode_common_interviews/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function(nums, target) {
    for(let i = 0; i < nums.length; i++){
        let difference = target - nums[i]
        for(let j = i + 1; j < nums.length; j++){
            console.log("DIFF: ", difference)
            console.log("I: ",i)cd
            console.log("J: ",j)
            if(difference == nums[j]){
                return [i,j]
            }
        }
    }
};

// console.log(twoSum([2,7,11,15], 9))
console.log(twoSum([3,2,4], 6))

File: grind_75/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        results = []
        difference_dict = {}
        i = 0
        difference_dict[target - nums[i]] = i
        for i in range(1, len(nums)):
            difference = nums[i]
            if difference not in difference_dict:
                difference_dict[target - difference] = i
            else:
                return [difference_dict[difference], i]
